staging/src/k8s.io/apiserver/pkg/endpoints/handlers/apply/managedfields.go

```diff
@@ -0,0 +1,103 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apply
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/structured-merge-diff/fieldpath"
+)
+
+// ManagedFields wraps around meta.Accessor and provides retrieval and update helpers for ManagedFields
+type ManagedFields interface {
+	// Directly get and set the VersionedFieldSet on the accessor
+	Get() map[string]metav1.VersionedFieldSet
+	Set(map[string]metav1.VersionedFieldSet)
+
+	// Decode takes the FieldSet from Accessor and decodes it into ManagedFields
+	Decode() (fieldpath.ManagedFields, error)
+	// Encode takes the ManagedFields, encodes them and stores them to the Accessor
+	Encode(fieldpath.ManagedFields) error
+}
+
+type managedFieldsAccessor struct {
+	metav1.Object
+}
+
+// ManagedFieldsAccessor creates the Accessor and returns it or errors
+func ManagedFieldsAccessor(from interface{}) (ManagedFields, error) {
+	accessor, err := meta.Accessor(from)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't get accessor: %v", err)
+	}
+	return &managedFieldsAccessor{Object: accessor}, nil
+}
+
+// Get the accessors ManagedFields
+// TODO: Always set them to nil here? (because we mostly do). Could be unexpected behavior.
+func (a *managedFieldsAccessor) Get() map[string]metav1.VersionedFieldSet {
+	return a.GetManagedFields()
+}
+
+// Set the accessors ManagedFields
+func (a *managedFieldsAccessor) Set(to map[string]metav1.VersionedFieldSet) {
+	a.SetManagedFields(to)
+}
+
+// Decode takes the FieldSet from Accessor and decodes it into ManagedFields
+func (a *managedFieldsAccessor) Decode() (fieldpath.ManagedFields, error) {
+	managed, err := DecodeManagedFields(a.Get())
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert managed fields from API: %v", err)
+	}
+	return managed, nil
+}
+
+// Encode takes the ManagedFields, encodes them and stores them to the Accessor
+func (a *managedFieldsAccessor) Encode(fields fieldpath.ManagedFields) error {
+	managedFields, err := EncodeManagedFields(fields)
+	if err != nil {
+		return fmt.Errorf("failed to convert back managed fields to API: %v", err)
+	}
+	a.Set(managedFields)
+	return nil
+}
+
+// TODO: ^ this or just plain helper functions? Or both?
+
+// DecodeObjectManagedFields creates an accessor for the the passed in runtime.Object, extracts its ManagedFields
+// and decodes them and sets them to nil for the accessor
+func DecodeObjectManagedFields(from interface{}) (fieldpath.ManagedFields, error) {
```

Comment:
that argument type should be `runtime.Object`

Reply:
I tried but was getting build errors because I used it with Unstructured. Just looked at it again after your comment and figured this step is no longer necessary. I was just adding the helpers for now, didn't look at what else to ditch so doing that now :+1: